staging/src/k8s.io/apimachinery/pkg/apis/meta/fuzzer/fuzzer.go

```diff
@@ -186,15 +186,15 @@ func v1FuzzerFuncs(codecs runtimeserializer.CodecFactory) []interface{} {
 			j.ResourceVersion = strconv.FormatUint(c.RandUint64(), 10)
 			j.UID = types.UID(c.RandString())
 
-			var sec, nsec int64
+			var sec, nsec uint32
 			c.Fuzz(&sec)
 			c.Fuzz(&nsec)
-			j.CreationTimestamp = metav1.Unix(sec, nsec).Rfc3339Copy()
+			j.CreationTimestamp = metav1.Unix(int64(sec), int64(nsec)).Rfc3339Copy()
```

Comment:
Both `sec` and `nsec` can be negative. Please see https://play.golang.org/p/fNwU8lLMYZB this example.

Reply:
https://play.golang.org/p/Mzo_G42gLQg

```
# min int32
parsing time: -2147483648, 0
parsed: 1901-12-13 20:45:52 +0000 UTC

# max int32
parsing time: 2147483647, 0
parsed: 2038-01-19 03:14:07 +0000 UTC

# min int64
parsing time: -9223372036854775808, 0
failed to parse time: parsing time "292277026596-12-04T15:30:08Z" as "2006-01-02T15:04:05Z07:00": cannot parse "77026596-12-04T15:30:08Z" as "-"
parsed: 0001-01-01 00:00:00 +0000 UTC

# max int64
parsing time: 9223372036854775807, 0
failed to parse time: parsing time "292277026596-12-04T15:30:07Z" as "2006-01-02T15:04:05Z07:00": cannot parse "77026596-12-04T15:30:07Z" as "-"
parsed: 0001-01-01 00:00:00 +0000 UTC
```

Reply:
In summary, I think negative int64 values can be valid Unix time, but there is some limit where any value beyond won't be parsable and will become "null" when we parse it https://github.com/kubernetes/kubernetes/blob/204ff6caeb8229e6d12b54809108e7f32c19502f/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/time.go#L93
The same applies to too large int64 values. 

We cannot change the Rfc3339Copy function because it's a public interface, and the issue only applies to tests that provide unreasonable unix time. 

I think an easy fix is to make [sec and nsec](https://github.com/kubernetes/kubernetes/blob/204ff6caeb8229e6d12b54809108e7f32c19502f/staging/src/k8s.io/apimachinery/pkg/apis/meta/fuzzer/fuzzer.go#L189) int32 instead of int64. The fuzz result will fall in the range of `1901-12-13`~`2038-01-19` (roughly, I didn't calculate `nsec`), which is probably good enough. 

A more accurate fix if you're interested in is to find out the upper and lower limits of valid sec and nsec, and make the fuzzer generating values within that range. 

After you fix the fuzzer, please re-run `hack/update-generated-api-compatibility-data.sh` with `UPDATE_COMPATIBILITY_FIXTURE_DATA=true` so that we have valid timestamps in our [test data](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/testdata/README.md).

Reply:
Thanks @roycaihw for the in depth details and the reproducible playground. I'll update the fixture data in a bit.